test/unittest/dds/participant/ParticipantTests.cpp

```diff
@@ -977,6 +1026,986 @@ TEST(ParticipantTests, SetListener)
+    ASSERT_EQ(participant->delete_subscriber(subscriber), ReturnCode_t::RETCODE_OK);
+
+    // Create the publisher and a data_writer that use the above topic
+    Publisher* publisher = participant->create_publisher(PUBLISHER_QOS_DEFAULT);
+    ASSERT_NE(publisher, nullptr);
+    DataWriter* data_writer = publisher->create_datawriter(topic, DATAWRITER_QOS_DEFAULT);
+    ASSERT_NE(data_writer, nullptr);
+    // Check that an error is given at trying to unregister a type that is been used by a data_writer
+    ASSERT_EQ(participant->unregister_type(type.get_type_name()), ReturnCode_t::RETCODE_PRECONDITION_NOT_MET);
+
+    ASSERT_EQ(publisher->delete_datawriter(data_writer), ReturnCode_t::RETCODE_OK);
+    ASSERT_EQ(participant->delete_publisher(publisher), ReturnCode_t::RETCODE_OK);
+
+    ASSERT_EQ(participant->delete_topic(topic), ReturnCode_t::RETCODE_OK);
+
+    // At this point, the type is not been used by any entity.
+    // Check that no errors result when an unused topic is unregistered
+    ASSERT_EQ(participant->unregister_type(type.get_type_name()), ReturnCode_t::RETCODE_OK);
+
+    // Remove the participant
+    ASSERT_EQ(DomainParticipantFactory::get_instance()->delete_participant(participant), ReturnCode_t::RETCODE_OK);
+}
+
+/*
+ * This test checks the new_remote_endpoint_discovered() DomainParticipant memeber function used in the
+ * STATIC discovery.
+ * 1. Check that the remote endpoint is not registered in a disabled participant.
+ * 2. Check that a WRITER remote endpoint is registered in an enabled participant.
+ * 3. Check that a READER remote endpoint is registered in an enabled participant.
+ */
+TEST(ParticipantTests, NewRemoteEndpointDiscovered)
+{
+    // Do not enable entities on creation
+    DomainParticipantFactory* factory = DomainParticipantFactory::get_instance();
+    DomainParticipantFactoryQos qos;
+    qos.entity_factory().autoenable_created_entities = false;
+    ASSERT_TRUE(factory->set_qos(qos) == ReturnCode_t::RETCODE_OK);
+
+    // Create a disabled participant
+    DomainParticipant* participant =
+            DomainParticipantFactory::get_instance()->create_participant(0, PARTICIPANT_QOS_DEFAULT);
+    ASSERT_NE(nullptr, participant);
+    ASSERT_FALSE(participant->is_enabled());
+
+    eprosima::fastrtps::rtps::GUID_t remote_endpoint_guid;
+    std::istringstream("72.61.75.6c.5f.73.61.6e.63.68.65.7a") >> remote_endpoint_guid;
+
+    // Check that the remote endpoint is not registered in a disabled participant
+    ASSERT_FALSE(participant->new_remote_endpoint_discovered(
+                remote_endpoint_guid, 1, eprosima::fastrtps::rtps::EndpointKind_t::WRITER));
+
+    // Enable the participant
+    EXPECT_EQ(ReturnCode_t::RETCODE_OK, participant->enable());
+    EXPECT_TRUE(participant->is_enabled());
+
+    // Check that a WRITER remote endpoint is registered in an enabled participant
+    ASSERT_FALSE(participant->new_remote_endpoint_discovered(
```

Comment:
Should not be this `ASSERT_TRUE`??

Reply:
The comments are wrong. Here we are testing the negative clauses.

Reply:
This comment has not beel fixed yet.